src/context.ts

```diff
@@ -67,15 +76,17 @@ export class Context {
     readonly message?: any,
   ) {}
 
-  getData(options: { requiredOnly: boolean } = { requiredOnly: false }) {
+  getData(options: { requiredOnly: IncludeOptionals } = { requiredOnly: false }) {
     const { optional } = this;
     const checks =
-      options.requiredOnly && optional
+      options.requiredOnly === true && optional
         ? [
             (value: any) => value !== undefined,
             (value: any) => (optional === 'null' ? value != null : true),
             (value: any) => (optional === 'falsy' ? value : true),
           ]
+        : options.requiredOnly === 'ignoreUndefined'
+        ? [(value: any) => value !== undefined]
```

Comment:
In this way we can manage to get a nice solution that also encourages developers to use right now the new api. In this way, internally we can still call getData() using includeOptionals (e.g. in context runner)
```suggestion

    // TODO(8.0.0): remove requiredOnly
    getData(options: {
        /**
        * It is ignored is `includeOptionals` is specified
        * @deprecated use `includeOptionals` instead
        */
        requiredOnly: boolean,
        includeOptionals: IncludeOptionals
    } = { requiredOnly: false }) {
        const { optional } = this;
        const includeOptionals: IncludeOptionals = includeOptionals !== undefined ? includeOptionals : !requiredOnly
        const checks = 
            !includeOptionals && optional
            ? [
                (value: any) => value !== undefined,
                (value: any) => (optional === 'null' ? value != null : true),
                (value: any) => (optional === 'falsy' ? value : true),
            ]
            : includeOptionals === 'ignoreUndefined' && optional
            ? [
                (value: any) => value !== undefined,
            ]
            : []
```